test/src/tests-identity.js

```diff
@@ -1655,13 +1843,46 @@ describe('identity', function() {
                 result = null;
             });
 
-            // for some reason this is returning -4 for a good identity.
-            validUserIdentities.forEach(function(goodIdentities) {
-                mParticle.Identity[identityMethod](goodIdentities, callback);
-                result.httpCode.should.equal(200);
-                result.body.mpid.should.equal(testMPID);
-                result = null;
-            });
+            // modify request have a unique signature but the same validation
+            // for invalid values
+            if (identityMethod !== 'modify') {
+                // for some reason this is returning -4 for a good identity.
+                validUserIdentities.forEach(function(goodIdentities) {
+                    mParticle.Identity[identityMethod](
+                        goodIdentities,
+                        callback
+                    );
+                    expect(
+                        result.httpCode,
+                        `valid ${identityMethod} httpCode`
+                    ).to.equal(200);
+                    console.warn(result.body);
+                    expect(
+                        result.body.mpid,
+                        `valid ${identityMethod} mpid `
+                    ).to.be.ok;
+                    expect(
+                        result.body.mpid,
+                        `valid ${identityMethod} mpid`
+                    ).to.equal(testMPID);
+                    result = null;
```

Comment:
add a comment above here saying that this needs to be set to null because the test is inside of a loop and needs to be reset before each test loop